.github/workflows/build-and-run.yml

```diff
@@ -29,23 +34,26 @@ jobs:
       matrix:
         os:
           - label: Linux
-            runner: ubuntu-latest
-            ctest_exclude: -E ".*($ALL_EXCLUDE)"
-          - runner: macos-latest
-            label: macOS
+          - label: macOS
+            runner: macos-latest
             ctest_exclude: -E ".*($ALL_EXCLUDE|io\.gdal\.raster|unwrap\.snaphu|background)"
             # io.gdal.raster and unwrap.snaphu tests fail due to known unresolved issues
             # with GDAL memory mapping on macOS.
             # background reader/writer test sometimes fails on macOS due to slow runtimes.
         deps:
           - label: Latest
-            env-file: environment.yml
-            extra-specs: ""
         build-type: [ Release, Debug ]
-
+        include:
+          - os:
+              label: Linux
+            deps:
+              label: Minimum Python supported
+              extra-specs: |
+                python=3.9
+            build-type: RelWithDebInfo
```

Comment:
Let's define a `build-type` env variable above and default it to this value.

That way, when we add additional min-reqs jobs, we won't have to copy-paste this line for each.